src/notification/sms.ts

```diff
@@ -10,42 +10,55 @@ if (config.notifications.phone.number && !config.notifications.email.username) {
 
 const [email, phone] = [config.notifications.email, config.notifications.phone];
 
+if (phone.carrier.length !== phone.number.length) {
+	logger.warn('✖ the number of carriers must match the number of phone numbers');
+}
+
 export function sendSms(link: Link, store: Store) {
-	if (phone.number) {
+	for (let i = 0; i < Math.max(phone.number.length, phone.carrier.length); i++) {
+		const currentNumber = phone.number[i];
+		const currentCarrier = phone.carrier[i];
+
+		if (!currentNumber) {
+			logger.error(`✖ ${currentCarrier} is not associated with a number`);
+			continue;
+		} else if (!currentCarrier) {
+			logger.error(`✖ ${currentNumber} is not associated with a carrier`);
+			continue;
+		}
```

Comment:
Since we are doing the check on sizing in `sms.ts:13-15`, I'm not sure if this is needed.

Reply:
Oh, I see now. It's because it still allows the user to have misconfigured phone config.